R/zchunk_LA121.oil.R

```diff
@@ -35,70 +35,145 @@ module_energy_LA121.oil_DISABLED <- function(command, ...) {
-    #
-    #
-    # NOTE: there are 'match' calls in this code. You probably want to use left_join_error_no_match
-    # For more information, see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses vecpaste
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # NOTE: This code uses repeat_and_add_vector
-    # This function can be removed; see https://github.com/JGCRI/gcamdata/wiki/Name-That-Function
-    # ===================================================
 
+    # Calculating energy inputs (gas) to unconventional oil production in the historical years
+    A21.globaltech_coef %>%
+      left_join(calibrated_techs, by = c("supplysector", "subsector", "technology", "minicam.energy.input")) %>%
+      filter(sector != "NA") %>%
+      select(-sector, -fuel, -calibration, -secondary.output) -> L121.globaltech_coef
+
+    L121.globaltech_coef %>%
+      gather(year, value, -supplysector, -subsector, -technology, -minicam.energy.input) %>%
+      mutate(year = as.integer(year))  %>%
+    # Adding empty historical years to fill in with interpolation
+        complete(year = unique(c(HISTORICAL_YEARS, year)),
+                 nesting(supplysector, subsector, technology, minicam.energy.input)) %>%
+        arrange(year) %>%
+        group_by(technology, subsector, supplysector, minicam.energy.input)  %>%
+    # Interpolate to fill in missing globaltech_coef historical years
+        mutate(value = approx_fun(year, value)) %>%
+        left_join(distinct(calibrated_techs), by = c("supplysector", "subsector", "technology", "minicam.energy.input")) %>%
+        select(supplysector, subsector, technology, minicam.energy.input, year, value, sector, fuel) -> L121.globaltech_coef_interp
+
+    # Energy inputs = production times fuel IO coef
+    L111.Prod_EJ_R_F_Yh %>%
+      inner_join(L121.globaltech_coef_interp %>%
+                  rename(value_coef=value), by = c("sector", "fuel", "year")) %>%
+      select(GCAM_region_ID, sector, fuel, year, value, value_coef ) %>%
+      left_join(L121.globaltech_coef_interp %>%
+                  rename(value_coef_gas=value), by = c("sector", "year")) %>%
+      rename(fuel = fuel.y) %>%
+      mutate(value = value * value_coef_gas) %>%
+      select(GCAM_region_ID, sector, fuel, year, value) -> L121.in_EJ_R_unoil_F_Yh
+
+    # Downscaling unconventional oil consumption shares by GCAM 3.0 region to countries
+    product_filters <- IEA_product_rsrc %>% filter(resource == "crude oil")
+    product_filters <- unique(product_filters$PRODUCT)
+
+    L100.IEA_en_bal_ctry_hist %>%
+      filter(FLOW == "TPES", PRODUCT %in% product_filters) %>%
+      filter(year == max(HISTORICAL_YEARS)) %>%
+      select(iso, value) %>%
+      filter(value != "NA") %>%
+      group_by(iso) %>%
+      summarise(value = sum(value)) %>%
+      left_join(iso_GCAM_regID %>%
+                  select(region_GCAM3, iso), by="iso")-> L121.TPES_ktoe_ctry_oil_Yf
```

Comment:
Be,  Jill and I talked with Page to verify that this is not an instance where there is country-level data readily available to build up the GCAM 3 data that is being used.  Therefore we will leave it as is but add it as an issue for the future rather than implement an old version/ new version switch.